### PR TITLE
Add Medaka module

### PR DESCRIPTION
**Pull request by avantonder**
Add Medaka module. Note, for now the conda package isn't being found.

## PR checklist

Closes https://github.com/nf-core/modules/issues/824

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: modules/medaka/main.nf [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    conda (params.enable_conda ? "bioconda::medaka=1.4.4" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/medaka:1.4.4--py38h130def0_0"
+    } else {
+        container "quay.io/biocontainers/medaka:1.4.4--py38h130def0_0"
+    }
+
+    input:
+    tuple val(meta), path(reads), path(assembly)
```
Comment (FriederikeHanssen) [inline modules/medaka/main.nf:22]: is this one assembly per sample or is it the same across all samples? In that case I would suggest something like:

```
tuple val(meta), path(reads)
path(assembly)

File: tests/modules/medaka/main.nf [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+nextflow.enable.dsl = 2
+
+include { MEDAKA } from '../../../modules/medaka/main.nf' addParams( options: [:] )
+
+workflow test_medaka {
+    
+    input = [ [ id:'test', single_end:true ], // meta map
+              file(params.test_data['sarscov2']['nanopore']['test_fastq_gz'], checkIfExists: true),
+              fasta = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
```
Comment (FriederikeHanssen) [inline tests/modules/medaka/main.nf:11]:
```suggestion
             file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
```

**Review by FriederikeHanssen — CHANGES_REQUESTED** [review]

File: modules/medaka/main.nf (hunk 1 above)
Comment (d4straub) [reply modules/medaka/main.nf:22]: one assembly per sample, so keep as is

File: modules/medaka/main.nf [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    output:
+    tuple val(meta), path("*_polished_genome.fa.gz") , emit: polished_assembly
+    path "versions.yml"                              , emit: versions
```
Comment (drpatelh) [inline modules/medaka/main.nf:26]:
```suggestion
    tuple val(meta), path("*_polished_genome.fa.gz"), emit: assembly
    path "versions.yml"                             , emit: versions
```

File: modules/medaka/main.nf [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    script:
+    def prefix = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    """
+    medaka_consensus \\
+        -t ${task.cpus} \\
+        $options.args \\
+        -i ${reads} \\
+        -d ${assembly} \\
```
Comment (drpatelh) [inline modules/medaka/main.nf:35]:
```suggestion
        -t $task.cpus \\
        $options.args \\
        -i $reads \\
        -d $assembly \\
```

File: modules/medaka/main.nf [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        -o ./
+
+    mv consensus.fasta ${prefix}_polished_genome.fa
```
Comment (drpatelh) [inline modules/medaka/main.nf:38]:
```suggestion
    mv consensus.fasta ${prefix}.polished.genome.fa
```

File: modules/medaka/main.nf [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+    gzip -n ${prefix}_polished_genome.fa
```
Comment (drpatelh) [inline modules/medaka/main.nf:40]:
```suggestion
    gzip -n ${prefix}.polished.genome.fa
```

File: modules/medaka/meta.yml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - versions:
+      type: file
+      description: File containing software versions
+      pattern: "versions.yml"
+  - polished_assembly:
```
Comment (drpatelh) [inline modules/medaka/meta.yml:41]:
```suggestion
  - assembly:
```

File: modules/medaka/meta.yml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+      type: file
+      description: Polished genome assembly
+      pattern: "*_polished_genome.fa.gz"
```
Comment (drpatelh) [inline modules/medaka/meta.yml:44]:
```suggestion
      pattern: "*.polished.genome.fa.gz"
```

File: tests/modules/medaka/test.yml [hunk 9]
```diff
@@ -0,0 +1,7 @@
+- name: medaka test_medaka
+  command: nextflow run ./tests/modules/medaka -entry test_medaka -c tests/config/nextflow.config
+  tags:
+    - medaka
+  files:
+    - path: output/medaka/test_polished_genome.fa.gz
```
Comment (drpatelh) [inline tests/modules/medaka/test.yml:6]:
```suggestion
    - path: output/medaka/test.polished.genome.fa.gz
```

File: modules/medaka/meta.yml [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - versions:
+      type: file
+      description: File containing software versions
+      pattern: "versions.yml"
+  - assembly:
+      type: file
+      description: Polished genome assembly
+      pattern: "*.polished.genome.fa.gz"
```
Comment (drpatelh) [inline modules/medaka/meta.yml:44]:
```suggestion
      pattern: "*.fa.gz"
```

File: tests/modules/medaka/main.nf [hunk 11]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { MEDAKA } from '../../../modules/medaka/main.nf' addParams( options: [:] )
```
Comment (drpatelh) [inline tests/modules/medaka/main.nf:5]:
```suggestion
include { MEDAKA } from '../../../modules/medaka/main.nf' addParams( options: [suffix:'.polished.genome'] )
```

**Pull request merged**